backend/src/Controllers/voterRolls.controller.ts

```diff
@@ -44,7 +44,14 @@ const addElectionRoll = async (req: any, res: any, next: any) => {
             actor: req.user.email,
             timestamp: Date.now(),
         }]
-        const newElectionRoll = await ElectionRollModel.submitElectionRoll(req.election.election_id, req.body.VoterIDList, false, ElectionRollState.approved, history, req, `User adding Election Roll??`);
+        const rolls: ElectionRoll[] = req.body.VoterIDList.map((id:string) => ({
+            election_id: req.election.election_id,
+            voter_id: id,
+            submitted: false,
+            state: ElectionRollState.approved,
+            history: history,
```

Comment:
How we store the history list could be worth more discussion. 

Currently looks like we're storing a separate history list for each user in the roll list (1 list per user/election)? Here's a list of some other options

- 1 list for everything (a global history)
- 1 list per election
- 1 list per election server (for the future when our architecture spreads the load across multiple servers)
- 1 list per ballot
- 1 list per voter roll (current approach)

I haven't decided which approach I like best yet. "1 list per election" will make the auditing process cleaner (otherwise we're combing a bunch of smaller lists), but it also makes concurrency issues more likely. "1 list per election tenant" will be ideal for the hashing, and "1 list per voter roll" approach is ideal for voters looking up their ballot info

Reply:
Could you elaborate what you mean that "1 list per election will make auditing process cleaner?"

I'm imagining the two most common questions would be:
* is this voter approved
* why is this voter approved / not approved

which seem to be served by history per voter per election
(I'm also seeing "1 history per voter per election" the same as "1 list per ballot", but if there's a distinction here please LMK)

Reply:
I think the use for 1 list per election for an auditor to easily grab all the data and see for example all the voters that were rejected. You might also want voters to be able to see this data (with any sensitive data removed). But it should be easy enough to query all the histories and concatenate them if needed. 

My concern with 1 list is that it makes your concurrent modifications concern more likely. 

Another approach is we have a separate table for histories and just have 1 row per modification, but that sounds like overkill.

Reply:
Sure thing

| Could you elaborate what you mean that "1 list per election will make auditing process cleaner?"

I was thinking the auditor would want to see a full log of everything that happened in the election in one place

| I'm also seeing "1 history per voter per election" the same as "1 list per ballot", but if there's a distinction here please LMK

That one is more a question of where we store the history. There are actions like "vote counted", or "vote rejected", that are initiated by an admin rather than the voter, so maybe they make more sense attached to the ballot rather than the voter? Then again, the registration history would exist before the ballot, so maybe roll is the better option between the 2?

Reply:
I'm thinking voter roll history would have things like registration or changing state, ballot history would have things like if the ballot is invalidated or changed because the voter resubmitted or invalidated by an admin. I think we'll need to come up with a list of reasons why any of these would change/need history data and go from there.

Reply:
Ah, great thing to clarify:  are there reasons a vote would be rejected other than having to do with the voter registration?  Something about the ballot itself rather than about the voter?

What all information are we trying to capture in these histories?

For ballot, I assumed when the voter votes, and if they change their ballot.

For the registration, would want to see when the initial registration happened, when registration info is uploaded / changed (ex voter id etc)., as well as when approving / rejecting the voter.

What other events we need to capture?